step/errors.go

```diff
@@ -39,19 +41,33 @@ func wrapXcodebuildCommandError(cmd Printable, out string, err error) error {
 }
 
 func findXcodebuildErrors(out string) []string {
-	var errors []string
+	var errorLines []string
+	var nserrors []nserror.Error
 
 	scanner := bufio.NewScanner(strings.NewReader(out))
 	scanner.Split(bufio.ScanLines)
 	for scanner.Scan() {
 		line := scanner.Text()
 		if strings.HasPrefix(line, "error: ") {
-			errors = append(errors, line)
+			errorLines = append(errorLines, line)
+		} else if strings.HasPrefix(line, "Error ") {
+			e := nserror.New(line)
+			if e != nil {
```

Comment:
Related to my previous comment, if parsing the NSError fails, I feel we shouldn't drop the error entirely, I would add it to `errorLines`

Reply:
Good question. Currently, the parsed errors are 'human-readable' errors, while NSError is a structured error. 
It seemed all NSError has a regular error pair, by adding them regardless of the failure we would have some duplicated errors and some less human-readable errors (with extra context about the error).